src/main/java/vazkii/botania/common/Botania.java

```diff
@@ -45,11 +51,20 @@ public void preInit(FMLPreInitializationEvent event) {
 		thaumcraftLoaded = Loader.isModLoaded("Thaumcraft");
 		bcTriggersLoaded = ModAPIManager.INSTANCE.hasAPI("BuildCraftAPI|statements");
 		bloodMagicLoaded = Loader.isModLoaded("AWWayofTime"); // Psh, noob
+		
+		if(lightHelper==null)
+			lightHelper = new LightHelperVanilla();
 
 		proxy.preInit(event);
 	}
 
 	@EventHandler
+	@Optional.Method(modid = "easycoloredlights")
+	public void preInitColoredLights(FMLPreInitializationEvent event) {
+		lightHelper = new LightHelperColored();
```

Comment:
This was how they said to do it in the minecraftforge channel, and I can confirm that said code is called if and only if said mod is installed